plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/functions/tablechanges/TableChangesFunctionProcessor.java

```diff
@@ -0,0 +1,173 @@
+        Optional<Integer> changeOrdinalIndex = Optional.empty();
+        for (int columnIndex = 0; columnIndex < functionHandle.getColumns().size(); columnIndex++) {
+            IcebergColumnHandle column = functionHandle.getColumns().get(columnIndex);
+            if (column.getId() == DATA_CHANGE_TYPE_ID) {
+                changeTypeIndex = Optional.of(columnIndex);
+                delegateColumnMap[columnIndex] = -1;
+            }
+            else if (column.getId() == DATA_CHANGE_VERSION_ID) {
+                changeVersionIndex = Optional.of(columnIndex);
+                delegateColumnMap[columnIndex] = -1;
+            }
+            else if (column.getId() == DATA_CHANGE_TIMESTAMP_ID) {
+                changeTimestampIndex = Optional.of(columnIndex);
+                delegateColumnMap[columnIndex] = -1;
+            }
+            else if (column.getId() == DATA_CHANGE_ORDINAL_ID) {
+                changeOrdinalIndex = Optional.of(columnIndex);
+                delegateColumnMap[columnIndex] = -1;
+            }
+            else {
+                delegateColumnMap[columnIndex] = delegateColumnIndex;
+                delegateColumnIndex++;
+            }
+        }
+
+        this.split = split;
+        this.pageSource = icebergPageSourceProvider.createPageSource(
+                session,
+                functionHandle.getColumns(),
+                tableSchema,
+                partitionSpec,
+                PartitionData.fromJson(split.getPartitionDataJson(), partitionColumnTypes),
+                ImmutableList.of(),
+                DynamicFilter.EMPTY,
+                TupleDomain.all(),
+                split.getPath(),
+                0,
+                split.getLength(),
+                split.getFileSize(),
+                split.getFileRecordCount(),
+                split.getPartitionDataJson(),
+                split.getFileFormat(),
+                functionHandle.getNameMappingJson().map(NameMappingParser::fromJson));
+        this.delegateColumnMap = delegateColumnMap;
+        this.changeTypeIndex = changeTypeIndex;
+        this.changeVersionIndex = changeVersionIndex;
+        this.changeTimestampIndex = changeTimestampIndex;
+        this.changeOrdinalIndex = changeOrdinalIndex;
+    }
+
+    @Override
+    public TableFunctionProcessorState process()
+    {
+        while (!pageSource.isFinished()) {
+            Page dataPage = pageSource.getNextPage();
+            if (dataPage == null) {
+                continue;
+            }
+
+            Block[] blocks = new Block[delegateColumnMap.length];
+            for (int targetChannel = 0; targetChannel < delegateColumnMap.length; targetChannel++) {
+                int delegateIndex = delegateColumnMap[targetChannel];
+                if (delegateIndex != -1) {
+                    blocks[targetChannel] = dataPage.getBlock(delegateIndex);
+                }
+            }
+
+            changeTypeIndex.ifPresent(columnChannel ->
```

Comment:
I guess that these 4 things can be stored in a class field not to be calculated  for every  invocation of `process` function